### PR TITLE
contracts: Get rid of the dreaded `Other` error

### PR DESCRIPTION
**Pull request by athei**
This replaces the dreaded `Other` error by a `CodeRejected` error. This is the last occurrence of an error that isn't translated into a module specific error. It also prints a more detailed information on why it was rejected to the debug buffer (if only someone would inspect this buffer) and to the node console.

### CODE REVIEW

File: frame/contracts/src/lib.rs [hunk 1]
```diff
@@ -612,6 +612,10 @@ pub mod pallet {
 		/// or via RPC an `Ok` will be returned. In this case the caller needs to inspect the flags
 		/// to determine whether a reversion has taken place.
 		ContractReverted,
+		/// The contracts's code was found to be invalid during validation or instrumentation.
+		/// A more detailed error can be found on the node console with debug messages enabled
+		/// or in the debug buffer which is returned to RPC clients
```
Comment (HCastano) [inline frame/contracts/src/lib.rs:617]:
```suggestion
		/// or in the debug buffer which is returned to RPC clients.
```

File: frame/contracts/src/lib.rs [hunk 2]
```diff
@@ -612,6 +612,10 @@ pub mod pallet {
 		/// or via RPC an `Ok` will be returned. In this case the caller needs to inspect the flags
 		/// to determine whether a reversion has taken place.
 		ContractReverted,
+		/// The contracts's code was found to be invalid during validation or instrumentation.
```
Comment (HCastano) [inline frame/contracts/src/lib.rs:615]:
```suggestion
		/// The contract's code was found to be invalid during validation or instrumentation.
```

File: frame/contracts/src/lib.rs [hunk 3]
```diff
@@ -612,6 +612,10 @@ pub mod pallet {
...
+		/// A more detailed error can be found on the node console with debug messages enabled
```
Comment (HCastano) [inline frame/contracts/src/lib.rs:616]:
```suggestion
		/// A more detailed error can be found on the node console if debug messages are enabled
```

File: frame/contracts/src/tests.rs [hunk 4]
```diff
@@ -2899,3 +2899,29 @@ fn contract_reverted() {
 		assert_eq!(result.data.0, buffer);
 	});
 }
+
+#[test]
+fn code_rejected_error_works() {
```
Comment (HCastano) [inline frame/contracts/src/tests.rs:2904]: Is there any way to test that the debug buffer was written to correctly?

**Review by HCastano — APPROVED** [review]

File: frame/contracts/src/tests.rs (hunk 4 above)
Comment (athei) [reply frame/contracts/src/tests.rs:2904]: Yes. Added the check.

**Review by cmichi — APPROVED** [review]

**Pull request merged**